crates/compiler/parse/src/expr.rs

```diff
@@ -1190,26 +1190,27 @@ pub fn parse_single_def_assignment<'a>(
                 ));
             }
             Err(_) => {
-                // Unable to parse more defs, continue and return the first parsed expression as a stement
-                let empty_return =
-                    arena.alloc(Loc::at(first_loc_expr.region, Expr::EmptyDefsFinal));
-                let value_def = ValueDef::Body(
-                    arena.alloc(def_loc_pattern),
-                    arena.alloc(Loc::at(
-                        first_loc_expr.region,
-                        Expr::Defs(arena.alloc(defs), empty_return),
-                    )),
-                );
-                return Ok((
-                    progress_after_first,
-                    Some(SingleDef {
-                        type_or_value: Either::Second(value_def),
-                        region,
-                        spaces_before: spaces_before_current,
-                        spaces_after: &[],
-                    }),
-                    state_after_first_expression,
-                ));
+                todo!();
```

Comment:
This branch was never taken, going to add `unreachable!` after some logic investigation

Reply:
ah, actually we should just pass trough the returned error